staging/src/k8s.io/kubectl/pkg/util/resource/resource.go

```diff
@@ -35,156 +36,18 @@ func PodRequestsAndLimits(pod *corev1.Pod) (reqs, limits corev1.ResourceList) {
 	return podRequests(pod), podLimits(pod)
 }
 
-// podRequests is a simplified form of PodRequests from k8s.io/kubernetes/pkg/api/v1/resource that doesn't check
-// feature gate enablement and avoids adding a dependency on k8s.io/kubernetes/pkg/apis/core/v1 for kubectl.
 func podRequests(pod *corev1.Pod) corev1.ResourceList {
-	// attempt to reuse the maps if passed, or allocate otherwise
-	reqs := corev1.ResourceList{}
-
-	containerStatuses := map[string]*corev1.ContainerStatus{}
-	for i := range pod.Status.ContainerStatuses {
-		containerStatuses[pod.Status.ContainerStatuses[i].Name] = &pod.Status.ContainerStatuses[i]
-	}
-
-	for _, container := range pod.Spec.Containers {
-		containerReqs := container.Resources.Requests
-		cs, found := containerStatuses[container.Name]
-		if found && cs.Resources != nil {
-			if pod.Status.Resize == corev1.PodResizeStatusInfeasible {
-				containerReqs = cs.Resources.Requests.DeepCopy()
-			} else {
-				containerReqs = max(container.Resources.Requests, cs.Resources.Requests)
-			}
-		}
-		addResourceList(reqs, containerReqs)
-	}
-
-	restartableInitContainerReqs := corev1.ResourceList{}
-	initContainerReqs := corev1.ResourceList{}
-
-	for _, container := range pod.Spec.InitContainers {
-		containerReqs := container.Resources.Requests
-
-		if container.RestartPolicy != nil && *container.RestartPolicy == corev1.ContainerRestartPolicyAlways {
-			// and add them to the resulting cumulative container requests
-			addResourceList(reqs, containerReqs)
-
-			// track our cumulative restartable init container resources
-			addResourceList(restartableInitContainerReqs, containerReqs)
-			containerReqs = restartableInitContainerReqs
-		} else {
-			tmp := corev1.ResourceList{}
-			addResourceList(tmp, containerReqs)
-			addResourceList(tmp, restartableInitContainerReqs)
-			containerReqs = tmp
-		}
-		maxResourceList(initContainerReqs, containerReqs)
-	}
-
-	maxResourceList(reqs, initContainerReqs)
-
-	// Add overhead for running a pod to the sum of requests if requested:
-	if pod.Spec.Overhead != nil {
-		addResourceList(reqs, pod.Spec.Overhead)
-	}
-
+	reqs := resourcehelper.PodRequests(pod, resourcehelper.PodResourcesOptions{
```

Comment:
`PodResourcesOptions` There are a lot of parameters, I don't understand them in depth, but can we ensure that they are consistent with the original code? 🤔

Reply:
It is consistent as I confirmed. I'd appreciate if you experts can reconfirm it.